### ISSUE TITLE
data for ROILA-Net training

### ISSUE THREAD

**Issue by manhvd2103** [holzweber/palmprint-recognition#1]
Thanks so much for your repo !
I want to ask you how can I use the featureCreator.py file correctly and how to best label the dataset?

**Comment by holzweber** [comment]
Hello!

Good point - maybe I should go over the script once again and make it more clear!

I tried to do some changes to the script, which makes it a little bit simpler for now

But long term: (current commit https://github.com/holzweber/palmprint-recognition/commit/7971b2fd06d44d8484d75f40f30d744d48fbb24f)

So the idea is, that you can load an image of arbitrary height and width. 
I set the width and height in line 8 and 9 to 600x600, so the input images will be first resized to this format and also stored afterwards in this shape!
So if you want to change the output size of the annotated images, you should change it at this positions accordingly.

In the variable 'dir' (line 10), you should store the path, where all images are placed for annotation, the script will simply loop over them, displaying e.g. the current palm in a new window, where you can click inside, to plave a new landmark. In total, the script is planned to store 9 landmarks per hand image (since ROILaNet works with 9 x,y coordinates!).
If you are happy with your 9 landmarks a button on your keyboard e.g. space, the coordinates will be exported as a squeezed array inside a new csv file (exactly as the dataset, which is used in ROILaNet is doing it) -> the benefit is here, that I trained their network with my annotated Tongji image data, which gave quite good results!

The benefit of this quick script is, that you can easily throw away images inside the loop, when you are not happy with your landmark annotations -> just add a 10th landmark and press enter, or only add e.g. 7 landmarks and press enter -> only when exactly 9 landmarks are placed, the exporting will be done.

Also you will need two  output directories, since I do not create them in the script, so you need to create a csvLandmark directory (line 60) and a imgMask directory (line61)

How can you best label:
I would label the images, as the original paper proposed it, here is a small example, how I placed the landmarks (you need to click on this positions in the featureCreator-window, with your mouse)

![image](https://user-images.githubusercontent.com/48522299/169708652-550448a3-37c8-4103-b30b-d52d4b125f50.png)


I hope this helps
greetings, Chris

**Comment by manhvd2103** [comment]
Thanks you so much for this repo and this reply! It helped me a lot

**Comment by manhvd2103** [comment]
Hello holzweber !
Do you still keep the dataset for ROILA-Net model training? If so can you share it with me? I have done my dataset with about 3000 images but model training is overfitting. I also don't know what the reason is. Maybe due to little data? Hope you can help me!

**Comment by holzweber** [comment]
Hello!

Initially, ROILaNet was trained with the NTU Palmprints from the Internet (NTU-PI-v1) dataset - there you can find already annotated images (coordinates are in matlab files, but you can easily extract them). Based on this, I also annotated like 1000 additional images, randomly chosen from the Tongji dataset, and added them, which gave quite nice results for the spatial transformer network.

So I suggest, you apply for the dataset here: https://github.com/matkowski-voy/Palmprint-Recognition-in-the-Wild
They responded pretty quickly and sent a link to the dataset!

Greetings,
Chris

**Comment by manhvd2103** [comment]
Hello!
Sorry for the delay in commenting
I followed the instructions and emailed a few days ago but no response yet. I don't know if they're really active?

**Comment by czrb** [comment]
> hello！do you get the DataSet？I email them，too.

**Comment by holzweber** [comment]
Hey guys. I see some of you having trouble with getting the orignal dataset used in the paper. However, you actually can use my feature extractor to create your own dataset with available hand image databases in the internet.

I also see that in some files my code can be improved and I hope I can update some lines to make them more clear if I have more time beside my master's thesis !

Let me know if you have any questions and please open a new issue here (pls do not mail me  :D )
